ledger/src/blockstore.rs

```diff
@@ -770,6 +770,47 @@ impl Blockstore {
         recovered_data_shreds
     }
 
+    /// The main helper function that performs the shred insertion logic
+    /// and updates necessary meta-data.
+    ///
+    /// This function updates the following column families:
+    ///   - [`cf::DeadSlots`]: mark a shred as "dead" if its meta-data indicates
```

Comment:
nit: Marks the slot as dead if 
1) We get a new shred `N` marked as the last shred in the slot `S`, but `N.index()` is less than the current `slot_meta.received` for slot `S`
2) The slot is not currently full

This means there's an alternate version of this slot. See `check_insert_data_shred` for more details